sql.go

```diff
@@ -15,13 +16,25 @@ import (
 	"github.com/ydb-platform/ydb-go-sdk/v3/trace"
 )
 
-var d = &sqlDriver{connectors: make(map[*xsql.Connector]*Driver)} //nolint:gochecknoglobals
+var (
+	d          = &sqlDriver{connectors: make(map[*xsql.Connector]*Driver)} //nolint:gochecknoglobals
+	dsnParsers = []func(dsn string) (opts []config.Option, connectorOpts []ConnectorOption, _ error){
+		xsql.Parse,
+	}
+)
 
 func init() { //nolint:gochecknoinits
 	sql.Register("ydb", d)
 	sql.Register("ydb/v3", d)
 }
 
+// RegisterSqlOpenDsnParser registers DSN parser for sql.Open driver constructor
+//
+// Experimental: https://github.com/ydb-platform/ydb-go-sdk/blob/master/VERSIONING.md#experimental
+func RegisterSqlOpenDsnParser(parser func(dsn string) (opts []config.Option, connectorOpts []ConnectorOption, _ error)) {
```

Comment:
What about return `struct {Opts []config.Option, ConnectorOpts []ConnectorOption, RemoveID int}` for allow add parsed option in the future?

`RemoveID` need for remove registered handler (good for tests for revert global changed state).